source/dub/packagemanager.d

```diff
@@ -30,23 +30,42 @@ import std.zip;
 /// packages.
 class PackageManager {
 	private {
-		Repository[LocalPackageType] m_repositories;
+		Repository[Path] m_repos;
```

Comment:
This is the fundamental change. There are now an unlimited number of repositories, so we index them by path instead of an `enum`. Indexing by path is a little scary, which is why the `public` interface of this class has contracts enforcing absolute paths and normalised paths before using them.